dotnet/src/SemanticKernel.Core/KernelExtensions.cs

```diff
@@ -573,6 +606,43 @@ public static IKernelBuilderPlugins AddFromObject(this IKernelBuilderPlugins plu
 
         return function.InvokeStreamingAsync<StreamingKernelContent>(kernel, arguments, cancellationToken);
     }
+
+    /// <summary>
+    /// Invokes a prompt specified via a prompt template and streams its results of type <typeparamref name="T"/>.
+    /// </summary>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="promptTemplate">Plain language definition of the prompt, using SK prompt template language</param>
+    /// <param name="arguments">The arguments to pass to the function's invocation, including any <see cref="PromptExecutionSettings"/>.</param>
+    /// <param name="templateFormat">Optional format of the template. Must be provided if a prompt template factory is provided</param>
+    /// <param name="promptTemplateFactory">The template factory to use to interpret <paramref name="promptTemplate"/>.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>An <see cref="IAsyncEnumerable{T}"/> for streaming the results of the function's invocation.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="kernel"/> is null.</exception>
+    /// <exception cref="ArgumentNullException"><paramref name="promptTemplate"/> is null.</exception>
+    /// <exception cref="ArgumentException"><paramref name="promptTemplate"/> is empty or composed entirely of whitespace.</exception>
+    /// <remarks>
+    /// The function will not be invoked until an enumerator is retrieved from the returned <see cref="IAsyncEnumerable{T}"/>
+    /// and its iteration initiated via an initial call to <see cref="IAsyncEnumerator{T}.MoveNextAsync"/>.
+    /// </remarks>
+    public static IAsyncEnumerable<T> InvokePromptStreamingAsync<T>(
+        this Kernel kernel,
+        string promptTemplate,
+        KernelArguments? arguments = null,
+         string? templateFormat = null,
+        IPromptTemplateFactory? promptTemplateFactory = null,
```

Comment:
Something wrong with formatting:
```suggestion
        string? templateFormat = null,
        IPromptTemplateFactory? promptTemplateFactory = null,
```